src/index.tsx

```diff
@@ -592,11 +647,14 @@ export async function validateServerFormData<T extends FormDefinition>(
 // with multiple values
 function getInputDefaultValue<T extends FormDefinition>(
   name: string,
+  type?: InputValidationAttrs["type"],
   serverFormInfo?: ServerFormInfo<T>,
   index?: number
 ) {
   let submittedValue = serverFormInfo?.submittedValues?.[name];
-  if (Array.isArray(submittedValue)) {
+  if (type === "checkbox") {
+    return undefined;
```

Comment:
checkboxes don't need a `defaultValue` assigned.  They need to be re-populated via `checked` so going to open a separate issue for that.  Added this so we don't complain about "index" currently.

Reply:
https://github.com/brophdawg11/remix-validity-state/issues/51